actors/builtin/power/power_actor.go

```diff
@@ -563,9 +563,11 @@ func (a Actor) processDeferredCronEvents(rt Runtime) error {
 
 			cronEvents = append(cronEvents, epochEvents...)
 
-			err = st.clearCronEvents(store, epoch)
-			if err != nil {
-				return errors.Wrapf(err, "failed to clear cron events at %v", epoch)
+			if len(epochEvents) > 0 {
+				err = st.clearCronEvents(store, epoch)
+				if err != nil {
+					return errors.Wrapf(err, "failed to clear cron events at %v", epoch)
+				}
```

Comment:
From investigation we did together, `Delete` isn't idempotent and the error being returned: https://github.com/ipfs/go-hamt-ipld/blob/d4e402c05ab0aa9fa74d5d5f84533108b7f048da/hamt.go#L370

Not sure if we should be checking the returned error (:thinking:), checking for existence before deleting (:nauseated_face:), or mofiy go-hamt-ipld s.t. `Delete` is idempotent (:star:)

Reply:
cc @whyrusleeping @anorth what do y'all think?

Reply:
also, errors returned inside here are being ignored. Here is an example of how we have handled these in the past: https://github.com/filecoin-project/specs-actors/blob/master/actors/builtin/power/power_actor.go#L603

Reply:
🤢